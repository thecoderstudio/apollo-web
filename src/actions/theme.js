function changeTheme(theme) {
  return {
    type: 'CHANGE_THEME',
    theme
  };
};

export { changeTheme };
