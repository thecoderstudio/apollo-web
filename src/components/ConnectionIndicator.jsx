import React from 'react';
import styled, { withTheme } from 'styled-components';
import PropTypes from 'prop-types';

const propTypes = {
  connectionState: PropTypes.string.isRequired
};

const Indicator = styled.div`
  height: 12px;
  width: 12px;
  margin: 10px;
  border-radius: 50%;

  color: ${props => props.color};
  background-color: ${props => props.color};
`;

export function getColor(connectionState, theme) {
  switch (connectionState) {
    case 'connecting':
      return theme.connectingColor;
    case 'connected':
      return theme.connectedColor;
    case 'disconnected':
      return theme.disconnectedColor;
    default:
      return;
  }
}


function ConnectionIndicator(props) {
  return <Indicator className={props.className} color={getColor(props.connectionState, props.theme)} />;
}

ConnectionIndicator.propTypes = propTypes;

export default withTheme(ConnectionIndicator);
