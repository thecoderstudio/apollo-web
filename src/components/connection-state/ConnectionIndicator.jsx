import React from 'react';
import styled, { withTheme } from 'styled-components';
import PropTypes from 'prop-types';

const propTypes = {
  connectionState: PropTypes.string.isRequired
};

const Indicator = styled.div`
  min-height: 12px;
  min-width: 12px;
  margin: 10px;
  border-radius: 50%;

  color: ${props => props.color};
  background-color: ${props => props.color};
`;

export function getColor(connectionState, theme) {
  switch (connectionState) {
    case 'connecting':
      return theme.connectingColor;
    case 'connected':
      return theme.connectedColor;
    case 'disconnected':
      return theme.disconnectedColor;
    default:
      return null;
  }
}

function ConnectionIndicator(props) {
  return <Indicator className={props.className} color={getColor(props.connectionState, props.theme)} />;
}

ConnectionIndicator.propTypes = propTypes;

export default withTheme(ConnectionIndicator);
