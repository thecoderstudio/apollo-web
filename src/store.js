import { createStore } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2'
import storage from 'redux-persist/lib/storage';
import mapTransformer from './transformers/map';
import rootReducer from './reducers/root';

const persistConfig = {
  key: 'root',
  whitelist: ['auth', 'agent'],
  transforms: [mapTransformer({ whitelist: 'agent' })],
  storage
};

const persistedReducer = persistReducer(persistConfig, rootReducer);
export const store = createStore(persistedReducer);
export const persistor = persistStore(store);
