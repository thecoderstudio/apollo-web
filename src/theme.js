export const darkTheme = {
  name: "dark",
  black: "#181818",
  lightBlack: "#1B1B1B",
  white: "#F8F8F8",
  primary: "#02A3EE",
  accent: "#4D02EE",
  connectedColor: "#27AE60",
  disconnectedColor: "#C0392B",
  connectingColor: "#F39C12",
  error: "#ED3E2A"
};

function getBrowserPreferredTheme() {
  return darkTheme;
}

export const browserPreferredTheme = getBrowserPreferredTheme();
