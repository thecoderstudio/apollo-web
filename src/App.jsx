import React from 'react';
import { connect } from 'react-redux';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import styled, { createGlobalStyle, ThemeProvider } from 'styled-components';
import ProtectedRoute from './components/ProtectedRoute';
import NavBar from './components/NavBar';
import Notifications from './components/notifications/Notifications';
import Dashboard from './pages/Dashboard';
import Welcome from './pages/Welcome';
import AgentDetail from './pages/AgentDetail';
import TerminalPage from './pages/TerminalPage';
import Admin from './pages/Admin';
import NotFound from './pages/NotFound';
import { darkTheme } from './theme';

const GlobalStyle = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css2?family=B612:wght@400;700&family=Roboto:wght@300;400&display=swap');

  html, body {
    margin: 0px;
    padding: 0px;
    height: 100%;
    width: 100%;
    font-weight: 500;
    color: ${(props) => props.theme.white};
    background-color: ${(props) => props.theme.black};
  }

  body {
    font-family: 'Roboto', sans-serif;
  }

  h1, h2, h3, h4, h5, h6 {
    font-family: 'B612', sans-serif;
  }
`;

const Content = styled.div`
  height: 100%;
  width: 100%;
`;

function App(props) {
  return (
    <div>
      <ThemeProvider theme={darkTheme}>
        <BrowserRouter>
          <Content>
            {props.authenticated && props.hasChangedPassword && <NavBar />}
            <Switch>
              <ProtectedRoute exact path="/" component={Dashboard} fallbackComponent={Welcome} />
              <ProtectedRoute exact path="/agent/:agentId/shell" component={TerminalPage} fallbackComponent={NotFound} />
              <ProtectedRoute exact path="/agent/:agentId" component={AgentDetail} fallbackComponent={NotFound} />
              <ProtectedRoute exact path="/admin" component={Admin} fallbackComponent={NotFound} role="admin" />
              <Route component={NotFound} />
            </Switch>
            <Notifications />
            <GlobalStyle />
          </Content>
        </BrowserRouter>
      </ThemeProvider>
    </div>
  );
}

export default connect(
  (state) => ({
    authenticated: state.authenticated,
    hasChangedPassword: state.currentUser.hasChangedInitialPassword
  })
)(App);
