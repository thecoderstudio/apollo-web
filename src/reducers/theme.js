import { browserPreferredTheme } from '../theme';

export default function themeReducer(state = { theme: browserPreferredTheme }, action) {
  switch (action.type) {
    case 'CHANGE_THEME':
      return { theme: action.theme };
    default:
      return state;
  }
}
