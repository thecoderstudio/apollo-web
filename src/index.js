import React from 'react';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react'
import { render } from 'react-dom';
import App from './App';
import { store, persistor } from './store';

render(
  <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <App />
    </PersistGate>
  </Provider>,
  document.getElementById('app')
);
