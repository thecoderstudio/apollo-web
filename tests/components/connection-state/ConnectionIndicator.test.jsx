import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import ConnectionIndicator, { getColor } from '../../../src/components/connection-state/ConnectionIndicator';
import { darkTheme } from '../../../src/theme';

function getComponent(state) {
  return renderer.create(
    <ThemeProvider theme={darkTheme} >
      <ConnectionIndicator connectionState={state} />
    </ThemeProvider>
  );
}

describe("ConnectionIndicator", () => {
  it("renders correctly", () => {
    const tree = getComponent("connected").toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Gets correct color", () => {
    let tree = getComponent("blanko").toJSON();
    expect(tree).toMatchSnapshot();
    tree = getComponent("disconnected").toJSON();
    expect(tree).toMatchSnapshot();
    tree = getComponent("connecting").toJSON();
    expect(tree).toMatchSnapshot();
    tree = getComponent("connected").toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("getColor", () => {
  it("returns the correct color for each connectionState", () => {
    expect(getColor('fake', darkTheme)).toBe(null);
    expect(getColor('connecting', darkTheme)).toBe(darkTheme.connectingColor);
    expect(getColor('connected', darkTheme)).toBe(darkTheme.connectedColor);
    expect(getColor('disconnected', darkTheme)).toBe(darkTheme.disconnectedColor);
  });
});
