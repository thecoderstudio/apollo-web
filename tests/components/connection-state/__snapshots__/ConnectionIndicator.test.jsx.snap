// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionIndicator Gets correct color 1`] = `
.c0 {
  min-height: 12px;
  min-width: 12px;
  margin: 10px;
  border-radius: 50%;
}

<div
  className="c0"
  color={null}
/>
`;

exports[`ConnectionIndicator Gets correct color 2`] = `
.c0 {
  min-height: 12px;
  min-width: 12px;
  margin: 10px;
  border-radius: 50%;
  color: #C0392B;
  background-color: #C0392B;
}

<div
  className="c0"
  color="#C0392B"
/>
`;

exports[`ConnectionIndicator Gets correct color 3`] = `
.c0 {
  min-height: 12px;
  min-width: 12px;
  margin: 10px;
  border-radius: 50%;
  color: #F39C12;
  background-color: #F39C12;
}

<div
  className="c0"
  color="#F39C12"
/>
`;

exports[`ConnectionIndicator Gets correct color 4`] = `
.c0 {
  min-height: 12px;
  min-width: 12px;
  margin: 10px;
  border-radius: 50%;
  color: #27AE60;
  background-color: #27AE60;
}

<div
  className="c0"
  color="#27AE60"
/>
`;

exports[`ConnectionIndicator renders correctly 1`] = `
.c0 {
  min-height: 12px;
  min-width: 12px;
  margin: 10px;
  border-radius: 50%;
  color: #27AE60;
  background-color: #27AE60;
}

<div
  className="c0"
  color="#27AE60"
/>
`;
